Sources/FoundationEssentials/FileManager/FileOperations.swift

```diff
@@ -903,6 +911,43 @@ enum _FileOperations {
     }
     #endif
 
+    #if os(WASI)
+    private static func _linkOrCopyFile(_ srcPtr: UnsafePointer<CChar>, _ dstPtr: UnsafePointer<CChar>, with fileManager: FileManager, delegate: some LinkOrCopyDelegate) throws {
+        var stat = stat()
+        guard lstat(srcPtr, &stat) == 0, !stat.isDirectory else {
+            // wasi-libc does not support FTS for now, so we don't support copying/linking
+            // directories on WASI for now.
+            throw CocoaError(.featureUnsupported)
```

Comment:
Similar comment as above - should we include the path here via `String(cString: srcPtr)`?